packages/shared-ux/chrome/navigation/__jest__/build_nav_tree.test.tsx

```diff
@@ -0,0 +1,738 @@
+      try {
+        expect(await findByTestId(/nav-item-group1.item1\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.item2\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.group1A\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.group1A.item1\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.group1A.group1A_1\s/)).toBeVisible();
+
+        // Click the last group to expand and show the last depth
+        (await findByTestId(/nav-item-group1.group1A.group1A_1\s/)).click();
+
+        expect(await findByTestId(/nav-item-group1.group1A.group1A_1.item1/)).toBeVisible();
+
+        expect(onProjectNavigationChange).toHaveBeenCalled();
+        const lastCall =
+          onProjectNavigationChange.mock.calls[onProjectNavigationChange.mock.calls.length - 1];
+        const [{ navigationTree }] = lastCall;
+        return navigationTree;
+      } catch (e) {
+        errorHandler(type)(e);
+      }
+    };
+
+    // -- Default navigation
+    {
+      const renderResult = renderNavigation({
+        navTreeDef: {
+          body: [
+            {
+              type: 'navGroup',
+              id: 'group1',
+              defaultIsCollapsed: false,
+              children: [
+                {
+                  id: 'item1',
+                  title: 'Item 1',
+                  href: 'https://foo',
+                },
+                {
+                  id: 'item2',
+                  title: 'Item 2',
+                  href: 'https://foo',
+                },
+                {
+                  id: 'group1A',
+                  title: 'Group1A',
+                  defaultIsCollapsed: false,
+                  children: [
+                    {
+                      id: 'item1',
+                      title: 'Group 1A Item 1',
+                      href: 'https://foo',
+                    },
+                    {
+                      id: 'group1A_1',
+                      title: 'Group1A_1',
+                      children: [
+                        {
+                          id: 'item1',
+                          title: 'Group 1A_1 Item 1',
+                          href: 'https://foo',
+                        },
+                      ],
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        },
+        onProjectNavigationChange,
+      });
+
+      const navigationTree = await runTests('treeDef', renderResult);
+
+      expect(navigationTree).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "children": Array [
+              Object {
+                "children": undefined,
+                "deepLink": undefined,
+                "href": "https://foo",
+                "id": "item1",
+                "isActive": false,
+                "isGroup": false,
+                "path": Array [
+                  "group1",
+                  "item1",
```

Comment:
Ideally we would not have these giant snapshots, but instead assert specifically for what we are testing.

If we want to test all items are present, could be something like:

```suggestion
      expect(navigationTree[0][0].id).toBe('item1');
      expect(navigationTree[0][1].id).toBe('item1');
      // ...
```

Reply:
Agree, I used to have even more of those. Decided to keep just a few here and use the technique you mention or sometimes better with `toMatchObject` (e.g. https://github.com/elastic/kibana/pull/169915/files#diff-3b7f2e9eb53f8631e5b02aaf0b613281f45a7c08df9f71619a4390138f6969a2R44) where I pick the fields that are relevant for the test